mock iotexapi

chenyanchen opened the pull request.

chenyanchen committed:
mock iotexapi

chenyanchen commented:
@yutongp

yutongp commented on misc/scripts/mockgen.sh:
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+mkdir -p ./test/mock
```
Let's put it under golang/iotexapi/mock_iotexapi

yutongp commented on Makefile:
```diff
@@ -22,3 +22,7 @@ gogen:
 	@protoc -I. -I./proto/types --go_out=plugins=grpc:${GOPATH}/src ./proto/api/*
 	@protoc -I. --grpc-gateway_out=logtostderr=true:${GOPATH}/src ./proto/api/*
+
+.PHONY: mockgen
```
Maybe add a `make gen` command run both gogen and mockgen?

chenyanchen committed:
integrate make cmd

1. move mock files to golang/iotexapi/
2. integrate make cmd and update README.md
